csrc/ir/utils.cpp

```diff
@@ -198,6 +198,16 @@ Expr* replaceValInExprInputs(Expr* expr, Val* reference, Val* substitute) {
       expr, reference, substitute);
 }
 
+void replaceVal(Val* old_val, Val* new_val) {
```

Comment:
nitpick: We have `replaceValue`, which is a more general version? That one does require a hash map for the replacement.... maybe we can switch the implementation to use that for consistency.

Reply:
Some reworking or renaming is in order here for sure. The problem is that while `replaceValue` is more general in the sense that it can do multiple replacements in a single traversal, it doesn't replace any vals in `fusion->outputs()`. This utility is meant to replace it in all uses and also in the outputs list. It could use a better name though at the very least.

Reply:
I think this needs to be more explicit and clear. For example, why just replacing outputs but not inputs? Would it make more sense to name it like `replaceAllExprInputsAndOutputs`?

Reply:
Went with the verbose but explicit `replaceValInAllExprInputsAndFusionOutputs`.